actors/evm/src/interpreter/instructions/context.rs

```diff
@@ -106,12 +106,20 @@ pub fn block_number(
     Ok(U256::from(system.rt.curr_epoch()))
 }
 
+/// EIP-4399: DIFFICULTY -> PREVRANDAO
 #[inline]
-pub fn difficulty(
+pub fn prevrandao(
     _state: &mut ExecutionState,
-    _system: &System<impl Runtime>,
+    system: &mut System<impl Runtime>,
 ) -> Result<U256, StatusCode> {
-    Ok(U256::zero())
+    // NOTE: Beacon randomness return value is expected to be `value > 2^64`.
+    // NOTE: Entropy must be consistent between contracts.
+    // NOTE: Getting randomness from beacon is expected to be deterministic for any given (previous) epoch.
+    // NOTE: EVM uses previous RANDAO value since the _current_ RANDAO for them runs as a smart contract on current state
```

Comment:
Write down things you find important to communicate to future readers, not necessarily things _you've_ learned.

Otherwise, readers are going to wonder why these things are important and get a bit confused.

Reply:
I've removed one of the notes and re-worded and moved the others to what I think will be important for a future reader to be aware of.

The extra long note I believe should stay since its a fair divergence from what the opcode in EVM does and the note provides context as to why it may seem divergent behavior.

Reply:
Yeah, that's much better. Although I still have no idea what the first comment is about.

My point here was more: you're writing a letter to future readers (possibly yourself).

Reply:
First comment is about preserving the expectation that PREVRANDAO less than `2^64` should only be returned when running in ETH1, so that in smart contracts it can be used to determine if it is running pre or post ETH2 merge. 

The note is there to say that beacon randomness (and therefore PREVRANDAO) is expected to always indicate that the smart contract is running post merge.